### ISSUE TITLE
Need to be able to lookup teams based on the list of names considering that not all of them may exists yet

### ISSUE THREAD

**Issue by dee-kryvenko** [env0/terraform-provider-env0#982]
**Is your feature request related to a problem? Please describe.**

Consider SSO enabled Env0 org. Env0 to my knowledge does not use SCIM at the moment, so the Teams are being populated to Env0 only when a first member of corresponding IdP group log in for the first time. Until then the team does not exists.

Consider a TF module that's used to create Env0 projects and assign permissions to corresponding teams. Such a module would accept a list of group names, which may include groups names that doesn't yet exists as Env0 Teams.

Trying to do something like this in that case would fail:

```
data "env0_team" "compartment_admins" {
  for_each = toset(local.okta_admin_groups)
  name     = each.key
}
```

The error would be something like

```
╷
│ Error: Could not find an env0 team with name <redacted>
│ 
│   with data.env0_team.compartment_admins["<redacted>"],
│   on env0_project_policy.tf line 1, in data "env0_team" "compartment_admins":
│    1: data "env0_team" "compartment_admins" {
│ 
╵
```

**Describe the solution you'd like**

There needs to be a way to account for a fact that the team might not exist yet, so the code would not fail and process other groups that do. Maybe `env0_teams` data source needs to be able to provide some kind of filter as an input, but there needs to be corresponding API endpoint otherwise if TF providers would do the filtering client-side - that wouldn't actually solve anything.

**Describe alternatives you've considered**

As a workaround I considered the following approach

```
data "env0_teams" "all_teams" {}

data "env0_team" "compartment_admins" {
  for_each = toset([
    for group in data.env0_teams.all_teams.names : group
    if contains(local.okta_admin_groups, group)
  ])
  name = each.value
}
```

The downside is that `env0_teams` data source trying to get a list of all teams in the org, which is excessive and unnecessary. Additionally, it fails agains another issue described in https://github.com/env0/terraform-provider-env0/issues/981

**Additional context**

**Comment by TomerHeber** [comment]
Hi @dee-kryvenko 

I'm trying to understand the request. And what would be the best solution for your requirements.
Please note that teams pagination is resolved.
We could add a filter to data "env0_teams" - would that help?

Thanks!

**Comment by dee-kryvenko** [comment]
Yes, I think that the ability to specify a filter would help, and data source must not fail if the result is empty list. It would just be empty list.

**Comment by liranfarage89** [comment]
@dee-kryvenko I wonder if now, after resolving #981 you still need the filter expression, because it's basically like the workaround you suggested but behind the scene. It should work, so I don't see a real value of encapsulating it behind a Resource's property.

cc @TomerHeber

**Comment by dee-kryvenko** [comment]
#981 is the workaround, this one is the actual issue. There's no reason to query backend for more than I need and having to filter on the client side. I have like 10k groups, even with pagination 100 per page - TF would make 100 requests where potentially only one was enough. Do it over and over across multiple environments during a drift detection.. just so inefficient.

**Comment by liranfarage89** [comment]
>There's no reason to query backend for more than I need and having to filter on the client side

@TomerHeber @yaronya IIUC with the current implementation in the [related pr](https://github.com/env0/terraform-provider-env0/pull/984) we should also implement a backend side... Otherwise the requests for all teams will remain.

**Comment by liranfarage89** [comment]
@TomerHeber

>The list of teams is now cached.

WDYM? isn't the team list fetched on every terraform "plan"?

**Comment by TomerHeber** [comment]
> @TomerHeber
> 
> >The list of teams is now cached.
> 
> WDYM? isn't the teams list fetched on every terraform "plan"?

Yes. But just once 
Before, it was called many times in the same plan, causing issues.

**Comment by liranfarage89** [comment]
Can you please explain more about it? Because I don't understand what did we change exactly. 
I just thought that we only added pagination (and the regex filter)
